src/dcqc/suites/suite_abc.py

```diff
@@ -0,0 +1,157 @@
+from __future__ import annotations
+
+from abc import ABC
+from collections.abc import Collection
+from itertools import chain
+from typing import Optional, Type, overload
+
+from dcqc.enums import TestStatus
+from dcqc.file import FileType
+from dcqc.mixins import SerializableMixin
+from dcqc.target import Target
+from dcqc.tests.test_abc import TestABC
+
+
+# TODO: Consider the Composite design pattern once
+#       we have higher-level QC suites
+class SuiteABC(SerializableMixin, ABC):
+    # Class attributes
+    file_type: FileType
+    add_tests: tuple[Type[TestABC], ...]
+    del_tests: tuple[Type[TestABC], ...]
+
+    # Instance attributes
+    type: str
+    target: Target
+    required_tests: set[str]
+    skipped_tests: set[str]
+
+    def __init__(
+        self,
+        target: Target,
+        required_tests: Optional[Collection[str]] = None,
+        skipped_tests: Optional[Collection[str]] = None,
+    ):
+        self.type = self.__class__.__name__
+        self.target = target
+
+        test_classes = self.list_test_classes()
+        test_names = set(test.__name__ for test in test_classes)
+
+        required_tests = required_tests or self._default_required_tests()
+        self.required_tests = set(required_tests).intersection(test_names)
+
+        skipped_tests = skipped_tests or list()
+        self.skipped_tests = set(skipped_tests).intersection(test_names)
+
+        self.tests = self.init_test_classes()
+        self._status = TestStatus.NONE
+
+    @classmethod
+    def list_test_classes(cls):
```

Comment:
Nit: Missing doc strings & type hinting

Lets think about which functions should return a list, which should a set.